File: examples/advanced-crud/src/main/java/org/example/advancedcrud/app/repository/UserRepository.java
package org.example.advancedcrud.app.repository;

import org.example.advancedcrud.app.entity.User;
import org.example.advancedcrud.utils.BaseRepository;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long>, JpaSpecificationExecutor<User> , BaseRepository {
    <T> Optional<T> findOneByUsername(
            String username,
            Class<T> type
    );

    boolean existsByUsername(String username);
    Optional<User> findUserByUsername(String username);

    void deleteUserByUsername(String username);
}

File: examples/basic-crud-kotlin/src/main/kotlin/org/example/basiccrudkotlin/app/repository/UserRepository.kt
package org.example.basiccrudkotlin.app.repository;

import org.example.basiccrudkotlin.app.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.data.rest.core.annotation.RepositoryRestResource
import org.springframework.data.rest.core.annotation.RestResource
import org.springframework.stereotype.Repository

@RepositoryRestResource(path = "users")
interface UserRepository : JpaRepository<User, Long>, JpaSpecificationExecutor<User> {
}